server/region_syncer/server.go

```diff
@@ -148,7 +166,21 @@ func (s *RegionSyncer) RunServer(regionNotifier <-chan *core.RegionInfo, quit ch
 // Sync firstly tries to sync the history records to client.
 // then to sync the latest records.
 func (s *RegionSyncer) Sync(stream pdpb.PD_SyncRegionsServer) error {
+	s.mu.RLock()
+	ctx := s.mu.serverCtx
+	s.mu.RUnlock()
+
+	if ctx == nil {
```

Comment:
When will it become nil?

Reply:
`RaftCluster` is not started.

Reply:
Why not move this check to server so that we can avoid the duplicated definition.